Update project_cv_application.md with React.StrictMode warning (#28348)
TheOdinProject/curriculum: react/states_and_effects/project_cv_application.md

<div class="lesson-content__panel" markdown="1">

<div class="lesson-note" markdown="1">

#### Double rendering

> While working on this project, you might notice that some parts of your code execute twice (e.g., console logs appearing doubled). This is due to React.StrictMode and is intended behavior. The next lesson will provide more details, but for now, be aware that you can refer to this resource if you encounter related issues: [StrictMode](https://react.dev/reference/react/StrictMode)

Reviewer: We can streamline the note by just linking the docs when we mention `React.StrictMode` itself. IMHO, this also puts a little more focus on not worrying about it as it'll be more relevant next lesson, as opposed to someone feeling they still need to dive deeper into understanding StrictMode at this point (they're very unlikely to run into the actual issues that StrictMode is designed to catch in this project).

```suggestion
While working on this project, you might notice that some parts of your code execute twice (e.g., console logs appearing doubled). This is due to [React.StrictMode](https://react.dev/reference/react/StrictMode) and is intended behavior. We will go through this more in the next lesson, so don't worry about it for now.
```
